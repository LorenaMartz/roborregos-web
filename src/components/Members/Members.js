import React, {Component} from 'react';
import MembersGrid from './MembersGrid/MembersGrid.js';
import MembersHeader from './MembersHeader/MembersHeader.js';
import MembersJoinUs from './MembersJoinUs/MembersJoinUs.js';
import Footer from 'components/Footer/Footer.js';
import './Members.css';

/** Component class of Members page. */
class Members extends Component {
  /**
 * Class constructor
 * @param {list} props List of member data.
 */
  constructor(props) {
    super(props);
    this.members = props.membersData.members;
  }

  /**
 * Renders Responsive view of Member's body page.
 * @return {renderized_component} Heder, grid, join us section and footer.
 */
  render() {
    document.title = 'RoBorregos | Members';
    const inactive = this.members.filter((member) =>
      member.status === 'inactive').sort((a, b) => b.id - a.id);
    const active = this.members.filter((member) =>
      member.status === 'active' ||
      member.status === 'comitee').sort((a, b) => a.id - b.id);

    return (
      <div className='members-container'>
        <MembersHeader />
        <MembersGrid active_members={ active } inactive_members={ inactive } />
        <MembersJoinUs />
        <Footer />
      </div>
    );
  }
}

export default Members;
