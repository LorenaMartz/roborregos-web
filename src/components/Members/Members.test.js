// @flow
import { getByTestId } from '@testing-library/react'
import React from 'react'
import { render, unmountComponentAtNode } from 'react-dom'
import { act } from 'react-dom/test-utils'
import renderer from 'react-test-renderer'

import Members from './Members'

// Mock data
const membersData = {
  members: [
    {
      id: 1,
      name: 'First',
      lastname: 'Member1',
      status: 'active',
      role: 'Software Development',
      class: '2015',
      semesters: '8',
      subtitle: '',
      degree: 'ISD',
      description: 'Mock member #1.',
      github: 'https://github.com',
      github_user: 'git_username',
      linkedin: 'https://www.linkedin.com',
      resume_link: '',
      tags: '',
      Email: 'mock@gmail.com',
    },
  ],
}

let container = null
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement('div')
  if (document.body != null) {
    document.body.appendChild(container)
  } else {
    expect(document.body).not.toEqual(null)
  }
})

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container)
  if (container != null) {
    container.remove()
  } else {
    expect(container).not.toEqual(null)
  }
  container = null
})

it('<Members> Renders correctly with Title Name', () => {
  act(() => {
    render(<Members membersData={membersData} />, container)
  })
  expect(document.title).toBe('RoBorregos | Members')
})

it('MemberÂ´s components render correctly', () => {
  act(() => {
    render(<Members membersData={membersData} />, container)
  })
  expect(getByTestId(container, 'members-join-us')).not.toEqual(null)
  expect(getByTestId(container, 'members-grid-container')).not.toEqual(null)
})

it('Matches snapshot', () => {
  const tree = renderer.create(<Members membersData={membersData} />).toJSON()
  expect(tree).toMatchSnapshot()
})
