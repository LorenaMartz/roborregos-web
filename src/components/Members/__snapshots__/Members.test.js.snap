// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<div
  className="members-container"
  data-testid="members-container"
>
  <div
    className="main-container"
    style={
      Object {
        "background": "linear-gradient( 45deg, #141213E6 10%, #005E69E6 100%), url(github_pattern.svg) top left/auto repeat fixed",
      }
    }
  >
    <div
      className="container-legend"
    >
      <h2
        className="title-text-banner"
      >
        Members
      </h2>
      <div
        className="main-text"
      >
        <p>
          RoBorregoâ€™s community is made by students with different skills in robotics, logistics and networking, all joined with a passion for exploring new technologies and sharing their knowledge with everybody.
          <br />
        </p>
      </div>
    </div>
    <div
      className="footer"
    >
      <p>
        Scroll down and meet the us!
        <br />
      </p>
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-angle-down fa-w-10 icon-btn"
        data-icon="angle-down"
        data-prefix="fas"
        focusable="false"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        role="img"
        style={
          Object {
            "color": "#00FFFA",
          }
        }
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
  </div>
  <div
    className="members-grid-container"
    data-testid="members-grid-container"
  >
    <div>
      <div
        className="grid-title"
        style={
          Object {
            "display": "none",
          }
        }
      >
        <h1
          className="grid-title-text"
        >
          
        </h1>
      </div>
      <ul
        className="MuiGridList-root members-grid"
        style={
          Object {
            "margin": -1.5,
          }
        }
      >
        <li
          className="MuiGridListTile-root members-grid-tile"
          id="members-grid-tile-1"
          onClick={[Function]}
          style={
            Object {
              "height": "auto",
              "padding": 1.5,
              "width": "25%",
            }
          }
        >
          <div
            className="MuiGridListTile-tile"
          >
            <div
              className="member-image-container"
            >
              <img
                alt="First"
                className="member-image"
                src="placeholder-rectangle.png"
              />
              <div
                className="member-image-content"
              >
                <p>
                  First Member1
                </p>
                <div
                  className="member-image-icon"
                >
                  <svg
                    aria-hidden="true"
                    className="svg-inline--fa fa-code fa-w-20 fa-1x "
                    data-icon="code"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    style={Object {}}
                    viewBox="0 0 640 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M278.9 511.5l-61-17.7c-6.4-1.8-10-8.5-8.2-14.9L346.2 8.7c1.8-6.4 8.5-10 14.9-8.2l61 17.7c6.4 1.8 10 8.5 8.2 14.9L293.8 503.3c-1.9 6.4-8.5 10.1-14.9 8.2zm-114-112.2l43.5-46.4c4.6-4.9 4.3-12.7-.8-17.2L117 256l90.6-79.7c5.1-4.5 5.5-12.3.8-17.2l-43.5-46.4c-4.5-4.8-12.1-5.1-17-.5L3.8 247.2c-5.1 4.7-5.1 12.8 0 17.5l144.1 135.1c4.9 4.6 12.5 4.4 17-.5zm327.2.6l144.1-135.1c5.1-4.7 5.1-12.8 0-17.5L492.1 112.1c-4.8-4.5-12.4-4.3-17 .5L431.6 159c-4.6 4.9-4.3 12.7.8 17.2L523 256l-90.6 79.7c-5.1 4.5-5.5 12.3-.8 17.2l43.5 46.4c4.5 4.9 12.1 5.1 17 .6z"
                      fill="currentColor"
                      style={Object {}}
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </li>
      </ul>
    </div>
    <div>
      <div
        className="grid-title"
        style={
          Object {
            "display": "block",
          }
        }
      >
        <h1
          className="grid-title-text"
        >
          RoBorregos Legacy
        </h1>
      </div>
      <ul
        className="MuiGridList-root members-grid"
        style={
          Object {
            "margin": -1.5,
          }
        }
      />
    </div>
  </div>
  <div
    className="members-join-us"
    data-testid="members-join-us"
  >
    <button
      className="members-join-us-button members-join-us-button-contain members-join-us-button-text"
      data-testid="members-join-us-button"
      onClick={[Function]}
      type="button"
    >
      Join the Team
    </button>
  </div>
  <div
    className="footer-container"
  >
    <div
      className="footer-row row"
    >
      <div
        className="col-logo col-lg-4"
      >
        <img
          alt="logo"
          className="footer-logo"
          src="white_logo.png"
        />
      </div>
      <div
        className="sitemap-container col-lg-4"
      >
        <div
          className="sitemap-link"
        >
          <span>
            <a
              className="sitemap-link"
              href="/"
            >
              Home
            </a>
            <br />
          </span>
        </div>
        <div
          className="sitemap-link"
        >
          <span>
            <a
              className="sitemap-link"
              href="/about"
            >
              About
            </a>
            <br />
          </span>
        </div>
        <div
          className="sitemap-link"
        >
          <span>
            <a
              className="sitemap-link"
              href="/members"
            >
              Members
            </a>
            <br />
          </span>
        </div>
        <div
          className="sitemap-link"
        >
          <span>
            <a
              className="sitemap-link"
              href="/contact"
            >
              Contact
            </a>
            <br />
          </span>
        </div>
        <div
          className="sitemap-link"
        >
          <span>
            <a
              className="sitemap-link"
              href="/candidates"
            >
              Candidates
            </a>
            <br />
          </span>
        </div>
        <div
          className="sitemap-link"
        >
          <span>
            <a
              className="sitemap-link"
              href="/projects"
            >
              Projects
            </a>
            <br />
          </span>
        </div>
      </div>
      <div
        className="col-lg-4"
      >
        <div
          className="row"
        >
          <div
            className="goback-button"
          >
            <a
              aria-disabled={false}
              className="MuiButtonBase-root MuiIconButton-root sitemap-link MuiIconButton-colorInherit"
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex={0}
            >
              <span
                className="MuiIconButton-label"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 8l-6 6 1.41 1.41L12 10.83l4.59 4.58L18 14z"
                  />
                </svg>
                <div
                  className="goback-text"
                >
                  Back to top
                </div>
              </span>
              <span
                className="MuiTouchRipple-root"
              />
            </a>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="row-socialMedia"
          >
            <a
              className="icon-link"
              href="https://www.instagram.com/roborregos/"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                style={
                  Object {
                    "fontSize": 40,
                  }
                }
                viewBox="0 0 24 24"
              >
                <path
                  d="M7.8 2h8.4C19.4 2 22 4.6 22 7.8v8.4a5.8 5.8 0 0 1-5.8 5.8H7.8C4.6 22 2 19.4 2 16.2V7.8A5.8 5.8 0 0 1 7.8 2m-.2 2A3.6 3.6 0 0 0 4 7.6v8.8C4 18.39 5.61 20 7.6 20h8.8a3.6 3.6 0 0 0 3.6-3.6V7.6C20 5.61 18.39 4 16.4 4H7.6m9.65 1.5a1.25 1.25 0 0 1 1.25 1.25A1.25 1.25 0 0 1 17.25 8 1.25 1.25 0 0 1 16 6.75a1.25 1.25 0 0 1 1.25-1.25M12 7a5 5 0 0 1 5 5 5 5 0 0 1-5 5 5 5 0 0 1-5-5 5 5 0 0 1 5-5m0 2a3 3 0 0 0-3 3 3 3 0 0 0 3 3 3 3 0 0 0 3-3 3 3 0 0 0-3-3z"
                />
              </svg>
            </a>
            <a
              className="icon-link"
              href="https://www.facebook.com/RoBorregos/"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                style={
                  Object {
                    "fontSize": 40,
                  }
                }
                viewBox="0 0 24 24"
              >
                <path
                  d="M5 3h14a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2m13 2h-2.5A3.5 3.5 0 0 0 12 8.5V11h-2v3h2v7h3v-7h3v-3h-3V9a1 1 0 0 1 1-1h2V5z"
                />
              </svg>
            </a>
            <a
              className="icon-link"
              href="https://github.com/RoBorregos/"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                style={
                  Object {
                    "fontSize": 35,
                  }
                }
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 .3a12 12 0 0 0-3.8 23.4c.6.1.8-.3.8-.6v-2c-3.3.7-4-1.6-4-1.6-.6-1.4-1.4-1.8-1.4-1.8-1-.7.1-.7.1-.7 1.2 0 1.9 1.2 1.9 1.2 1 1.8 2.8 1.3 3.5 1 0-.8.4-1.3.7-1.6-2.7-.3-5.5-1.3-5.5-6 0-1.2.5-2.3 1.3-3.1-.2-.4-.6-1.6 0-3.2 0 0 1-.3 3.4 1.2a11.5 11.5 0 0 1 6 0c2.3-1.5 3.3-1.2 3.3-1.2.6 1.6.2 2.8 0 3.2.9.8 1.3 1.9 1.3 3.2 0 4.6-2.8 5.6-5.5 5.9.5.4.9 1 .9 2.2v3.3c0 .3.1.7.8.6A12 12 0 0 0 12 .3"
                />
              </svg>
            </a>
            <div
              className="mark-text"
            >
              @2020 RoBorregos
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
